### ISSUE TITLE
Can't compile 

### ISSUE THREAD

**Issue by nonetrix** [whoshuu/cpr#607]
OS:
```
                   -`                    nonetrix@ArchBTW 
                  .o+`                   ---------------- 
                 `ooo/                   OS: Arch Linux x86_64 
                `+oooo:                  Host: B450 AORUS PRO WIFI 
               `+oooooo:                 Kernel: 5.13.7-zen1-1-zen 
               -+oooooo+:                Uptime: 7 hours, 44 mins 
             `/:-:++oooo+:               Packages: 1042 (pacman) 
            `/++++/+++++++:              Shell: bash 5.1.8 
           `/++++++++++++++:             Resolution: 1920x1200 
          `/+++ooooooooooooo/`           DE: Plasma 5.22.4 
         ./ooosssso++osssssso+`          WM: KWin 
        .oossssso-````/ossssss+`         Theme: Breeze Dark [Plasma], Adwaita [GTK3] 
       -osssssso.      :ssssssso.        Icons: [Plasma], Adwaita [GTK3] 
      :osssssss/        osssso+++.       Terminal: konsole 
     /ossssssss/        +ssssooo/-       Terminal Font: DejaVu Sans Mono 12 
   `/ossssso+/:-        -:/+osssso+-     CPU: AMD Ryzen 7 3700X (16) @ 3.600GHz 
  `+sso+:-`                 `.-/+oso:    GPU: NVIDIA GeForce GTX 1080 
 `++:.                           `-/+/   Memory: 5057MiB / 32052MiB 
 .`                                 `/
                                                                 
                                                                 
```

CMake error:
```
[ 70%] Built target libcurl
[ 77%] Built target cpr
Consolidate compiler generated dependencies of target discord
[ 78%] Built target discord
[ 95%] Built target zlib
Consolidate compiler generated dependencies of target gtest
[ 96%] Building CXX object _deps/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
In file included from /home/nonetrix/Project/C++/discord/_deps/googletest-src/googletest/src/gtest-all.cc:42:
/home/nonetrix/Project/C++/discord/_deps/googletest-src/googletest/src/gtest-death-test.cc: In function ‘bool testing::internal::StackGrowsDown()’:
/home/nonetrix/Project/C++/discord/_deps/googletest-src/googletest/src/gtest-death-test.cc:1301:24: error: ‘dummy’ may be used uninitialized [-Werror=maybe-uninitialized]
 1301 |   StackLowerThanAddress(&dummy, &result);
      |   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/home/nonetrix/Project/C++/discord/_deps/googletest-src/googletest/src/gtest-death-test.cc:1290:13: note: by argument 1 of type ‘const void*’ to ‘void testing::internal::StackLowerThanAddress(const void*, bool*)’ declared here
 1290 | static void StackLowerThanAddress(const void* ptr, bool* result) {
      |             ^~~~~~~~~~~~~~~~~~~~~
/home/nonetrix/Project/C++/discord/_deps/googletest-src/googletest/src/gtest-death-test.cc:1299:7: note: ‘dummy’ declared here
 1299 |   int dummy;
      |       ^~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [_deps/googletest-build/googletest/CMakeFiles/gtest.dir/build.make:76: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:441: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

CMake file:
```
cmake_minimum_required(VERSION 3.21.1)
include(FetchContent)

FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/whoshuu/cpr.git GIT_TAG f4622efcb59d84071ae11404ae61bd821c1c344b) #cpr v1.6.2
FetchContent_MakeAvailable(cpr)


project (discord)

add_executable(discord main.cpp)
target_link_libraries(discord PRIVATE cpr::cpr)
```

CMake version:
```
cmake version 3.21.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

Source:
```
#include <iostream>
#include <cpr/cpr.h>

int main() {
    cpr::Response r = cpr::Get(cpr::Url{"https://discord.com/api/v9/channels/xxxxxxxxxxxxxxx/messages"},
                      cpr::Authentication{"authorization", "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"},
                      cpr::Parameters{{"content", "Hello world!"}});
    std::cout << "Goodbye world!" << std::endl;
}
```

CMake command:
```
cmake .
```

Make command:
```
make -j16
```

This seems like an issue in the cpr code or my compiler, I am kinda new to this stuff so not exactly sure tbh, if you need more information or something just ask me, but I tried to give as much as possible.

**Comment by COM8** [comment]
Thanks for this really well formulated issue!
To me this looks like google-test is failing to compile on your system.
In case you do not need the tests to be build, try replacing
```cmake
FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/whoshuu/cpr.git GIT_TAG f4622efcb59d84071ae11404ae61bd821c1c344b) #cpr v1.6.2
FetchContent_MakeAvailable(cpr)
```
with:
```cmake
FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/whoshuu/cpr.git GIT_TAG f4622efcb59d84071ae11404ae61bd821c1c344b) #cpr v1.6.2
set(CPR_BUILD_TESTS OFF CACHE INTERNAL "" FORCE)
FetchContent_MakeAvailable(cpr)
```

In case this does not help, could you please provide the full `cmake .` output.

**Comment by nonetrix** [comment]
Should have also gave the GCC version, might help with fixing this
```
gcc (GCC) 11.1.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by nonetrix** [comment]
This was able to fix it closing issue

**Comment by COM8** [comment]
Hmmm...
I'm also using gcc 11 without any problems.

**Comment by nonetrix** [comment]
Could be the Arch version maybe? Sometimes they compile them with different options 

I'll try compiling GCC from source and see if changes anything

**Comment by COM8** [comment]
google/googletest#3219

Fixed in: 73f49c0d25bc21a67ad53478a101902d95dcf046
